Doc/c-api/init.rst

```diff
@@ -1464,6 +1470,29 @@ function. You can create and destroy them using the following functions:
    :c:func:`Py_FinalizeEx` will destroy all sub-interpreters that
    haven't been explicitly destroyed at that point.
 
+Interpreter ID Objects
```

Comment:
I'm giving this some attention so that we can get it merged soon.
[This section looks good so far](https://cpython-previews--112548.org.readthedocs.build/en/112548/c-api/init.html#interpreter-id-objects)